axelrod/tests/integration/test_tournament.py

```diff
@@ -29,12 +32,19 @@ def setUpClass(cls):
         ]
         cls.expected_outcome.sort()
 
-    def test_full_tournament(self):
-        """A test to check that tournament runs with all non cheating strategies."""
-        strategies = [strategy() for strategy in axelrod.strategies]
-        tournament = axelrod.Tournament(
-            name="test", players=strategies, game=self.game, turns=2, repetitions=1
-        )
+    @given(tournaments(
+        strategies=axelrod.short_run_time_strategies,
+        min_size=10,
+        max_size=30,
+        min_turns=2,
+        max_turns=40,
```

Comment:
There are some strategies that do weird things around 200 turns (I can't recall perfectly but I think that something like that might have even been the original motivation of this test). As we're limiting the number of strategies to 30 I don't think performance would be that bad if we let `max_turns=210` which just gives us a slightly more robust check.